src/rgw/driver/rados/rgw_dedup_utils.cc

```diff
@@ -658,9 +675,30 @@ namespace rgw::dedup {
         f->dump_unsigned("Set HASH", this->set_hash_attrs);
       }
 
+      if (this->skip_shared_tail_objs) {
```

Comment:
I assume f is a formatter type? The first param in that case should really be a variable (either json or XML, but probably json). And would be better to just call encode_json() without the existance check if these are optional<>